### PR TITLE
[AIRFLOW-3984] Add tests for WinRMHook

### PR DESCRIPTION
**Pull request by feluelle**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3984
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

- fix docs
- refactoring code

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

Add tests for
* test_get_conn_exists
* test_get_conn_missing_remote_host
* test_get_conn_error
* test_get_conn_from_connection
* test_get_conn_no_username
* test_get_conn_no_endpoint

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by Fokko** [comment]
I've restarted the failing tests.

**Comment by feluelle** [comment]
Thanks @Fokko :)

File: airflow/contrib/hooks/winrm_hook.py [hunk 1]
```diff
@@ -32,21 +32,18 @@ class WinRMHook(BaseHook):
...
+    :param ssh_conn_id: connection id from airflow Connections from where
+        all the required parameters can be fetched like username and password.
         Thought the priority is given to the param passed during init
     :type ssh_conn_id: str
     :param endpoint: When set to `None`, endpoint will be constructed like this:
         'http://{remote_host}:{remote_port}/wsman'
     :type endpoint: str
-    :param remote_host: Remote host to connect to.
-        Ignored if `endpoint` is not `None`.
+    :param remote_host: Remote host to connect to. Ignored if `endpoint` is not `None`.
```
Comment (OmerJog) [inline airflow/contrib/hooks/winrm_hook.py]: It's a bit unclear if you mean None value or string that contains None

**Review by Fokko — APPROVED** [review]
Thanks @feluelle

**Pull request merged**